goss-testing/scripts/k8s_kyverno_pods_running.sh

```diff
@@ -39,28 +39,59 @@ do
     esac
 done
 
-# Checks if all kyverno pods are running. There should be a total of 3 running pods.
+# Checks if all expected kyverno pods are running.
+# Component label and number of pods in running status should be:
+#   admission-controller: 3
+#   background-controller: 1
+#   cleanup-controller: 1
+#   reports-controller: 1
+
+check_running() {
+    pod_label=$1
+    pod_count=$2
+    running_pods=$(kubectl get pods -n kyverno --field-selector=status.phase=Running -l "app.kubernetes.io/component=${pod_label}" --no-headers | wc -l)
+    rc=$?
+    if [[ ${rc} -ne 0 ]]
+    then
+        # Split into two echo commands for code readability
+        echo -n "ERROR: Command pipeline failed (return code $rc): " 1>&2
+        echo "kubectl get pods -n kyverno --field-selector=status.phase=Running -l "app.kubernetes.io/component=${pod_label}" --no-headers | wc -l" 1>&2
+        echo "FAIL"
+        exit 10
+    elif [[ "${running_pods}" != "${pod_count}" ]]
+    then
+        if [[ ${print_results} -eq 1 ]]
+        then
+            echo "ERROR: ${running_pods} out of ${pod_count} ${pod_label} pods are running."
+            if [[ "${pod_label}" == "admission-controller" ]]
+            then
+                echo "For high availability the recommended Kyverno ${pod_label} replica count is ${pod_count}."
+                echo "Check the logs, restart Kyverno, and ensure that all ${pod_count} Kyverno ${pod_label} pods are running."
```

Comment:
```suggestion
                echo "Check the pod logs, do a rollout restart of 'deployment.apps/kyverno-admission-controller', and ensure that all ${pod_count} Kyverno ${pod_label} pods are running."
```
I think this instruction could be a bit more specific. I added my recommendation.